src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -441,6 +453,199 @@ try
 }
 CATCH_RETURN()
 
+// Routine Description:
+// - Returns whether the user set or updated any of the font features to be applied
+bool DxFontRenderData::DidUserSetFeatures() const noexcept
+{
+    return _didUserSetFeatures;
+}
+
+// Routine Description:
+// - Updates our internal map of font features with the given features
+// - NOTE TO CALLER: Make sure to call UpdateFont after calling this for the feature changes
+//   to take place
+// Arguments:
+// - features - the features to update our map with
+void DxFontRenderData::SetFeatures(const std::unordered_map<std::wstring_view, uint32_t>& features)
+{
+    // Populate the feature map with the standard list first
+    std::unordered_map<DWRITE_FONT_FEATURE_TAG, uint32_t> featureMap{

```

Comment:
Perhaps leave comments to the right of each item identifying what it is. I figure `kern` is kerning and `locl` is localized glyph substitution... but it might be nice to just leave the rest.

Reply:
Makes sense!